kpops/pipeline_generator/pipeline.py

```diff
@@ -125,127 +74,38 @@ def load_from_yaml(
-                    original_from_component = self.components.find(
-                        original_from_component_name
-                    )
-                    inflated_from_component = original_from_component.inflate()[-1]
-                    if inflated_from_component is not original_from_component:
-                        resolved_from_component_name = inflated_from_component.name
-                    else:
-                        resolved_from_component_name = original_from_component_name
-                    resolved_from_component = self.components.find(
-                        resolved_from_component_name
-                    )
-                    enriched_component.weave_from_topics(
-                        resolved_from_component.to, from_topic
+            if (
+                env_file := Pipeline.pipeline_filename_environment(path, config)
+            ).exists():
+                environment_components_content = load_yaml_file(
+                    env_file, substitution=ENV
+                )
+                if not isinstance(environment_components_content, list):
+                    raise TypeError(
+                        f"The pipeline definition {env_file} should contain a list of components"
                     )
-            elif self.components:
-                # read from previous component
-                prev_component = self.components.last
-                enriched_component.weave_from_topics(prev_component.to)
-            self.components.add(enriched_component)
-
-    def enrich_component(
-        self,
-        component: PipelineComponent,
-    ) -> PipelineComponent:
-        """Enrich a pipeline component with env-specific config and substitute variables
-
-        :param component: Component to be enriched
-        :type component: PipelineComponent
-        :returns: Enriched component
-        :rtype: PipelineComponent
-        """
-        component.validate_ = True
-        env_component_as_dict = update_nested_pair(
-            self.env_components_index.get(component.name, {}),
-            # HACK: Pydantic .dict() doesn't create jsonable dict
-            json.loads(component.json(by_alias=True)),
-        )
-
-        component_data = self.substitute_in_component(env_component_as_dict)
+            pipeline_components = PipelineComponentFactory(
+                component_list,
+                environment_components_content,
+                registry,
+                config,
+                handlers,
+            )
+            pipeline_components_list.append(pipeline_components.components)
 
-        component_class = type(component)
-        return component_class(
-            enrich=False,
-            config=self.config,
-            handlers=self.handlers,
-            **component_data,
+        pipeline_components = PipelineComponents(
+            **{"components": list(chain(*pipeline_components_list))}
         )
+        return Pipeline(pipeline_components)
```

Comment:
I believe this might not be in our best interest regarding the future of KPOps. Combining multiple pipelines into one large one has several drawbacks imo :(
- validation of DAG, now you have several completely independent subgraphs
- handling of pipeline name placeholders gets cumbersome
- sequential execution → slooow
- one long log for the entire thing, hard to dissect what belongs to which pipeline
additionally the burden of testing and maintenance and especially debugging gets more complex